src/io/import/data1D.h

```diff
@@ -24,6 +24,8 @@ class Data1DImportFileFormat : public FileAndFormat
 
     Data1DImportFileFormat(std::string_view filename = "", Data1DImportFormat format = Data1DImportFormat::XY, int xColumn = 1,
                            int yColumn = 2, int errorColumn = 0);
+    Data1DImportFileFormat(std::string_view filename, Data1DImportFileFormat::Data1DImportFormat format, std::optional<double> removeAverageFromX, std::optional<double> xMin, std::optional<double> xMax,
+                           int nPointsToRemove, int xColumn = 1, int yColumn = 2, int errorColumn = 0);
```

Comment:
Your new constructor might as well demand all the parameters at once.
```suggestion
                           int nPointsToRemove, int xColumn, int yColumn, int errorColumn);
```